src/main/java/us/abstracta/wiresham/ReloadService.java

```diff
@@ -0,0 +1,103 @@
+package us.abstracta.wiresham;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ReloadService implements Runnable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ReloadService.class);
+  private final VirtualTcpService service;
+  private final ExecutorService reloadExecutorService =
+      Executors.newSingleThreadExecutor(new ThreadFactoryBuilder()
+          .setNameFormat("Auto-Reload-Service-%d").build());
+  private final File configFile;
+  private boolean registered;
+
+  public ReloadService(VirtualTcpService service, File configFile) {
+    this.service = service;
+    this.configFile = configFile;
+  }
+
+  @Override
+  public void run() {
+    Path watchServicePath = getAbsolutFlowPath();
+    WatchService watchService;
+    try {
+      watchService = FileSystems.getDefault().newWatchService();
+      watchServicePath.register(watchService, StandardWatchEventKinds.ENTRY_MODIFY);
+      registered = true;
+    } catch (IOException e) {
+      LOG.error("Error while initializing Reload Service", e);
+      return;
+    }
+    while (true) {
+      try {
+        WatchKey take = watchService.take();
+
+        if (!take.isValid()) {
+          LOG.error("File {} deleted or corrupted, check file integrity",
+              Paths.get(watchServicePath.toString(), configFile.getName()));
+        }
+        for (WatchEvent<?> pollEvent : take.pollEvents()) {
+          if (pollEvent.kind().equals(StandardWatchEventKinds.ENTRY_MODIFY)
+              && pollEvent.context() instanceof Path) {
+
+            Path p = (Path) pollEvent.context();
+            if (Paths.get(watchServicePath.toString(), configFile.getName())
+                .equals(Paths.get(watchServicePath.toString(), p.toString()))) {
+              LOG.info("File was modified, flow will be restarted with new changes");
+              service.stop(VirtualTcpServiceMain.STOP_TIMEOUT_MILLIS);
```

Comment:
esto haria que se pierdan las sesiones /conexiones que tenes on the fly. Una opcion es hacer eso, otra es considerar solo los cambios para las nuevas conexiones. En cualquiera de los casos me parece que estaria bueno agregar algo en la opcion que indique que va a pasar eso (cada vez que se cambie el archivo, las conexiones existentes se van a cortar)? Buenisimo notificarlo en el log como info, pero capaz poner algun "warning" de ante mano?